scalafmt-core/shared/src/main/scala/org/scalafmt/rewrite/RedundantBraces.scala

```diff
@@ -108,24 +108,29 @@ class RedundantBraces(ftoks: FormatTokens) extends FormatTokensRewrite.Rule {
       val rt = ft.right
       new Token.LeftParen(rt.input, rt.dialect, rt.start)
     } // we will not keep it but will hint to onRight
-    ft.meta.rightOwner match {
-      case t: Term.Function if t.tokens.last.is[Token.RightBrace] =>
-        if (!okToRemoveFunctionInApplyOrInit(t)) null
-        else if (okToReplaceFunctionInSingleArgApply(t)) replace
-        else removeToken
-      case t: Term.Block =>
-        if (okToReplaceBlockInSingleArgApply(t)) replace
-        else if (processBlock(t)) removeToken
-        else null
-      case _: Term.Interpolate =>
-        if (processInterpolation) removeToken else null
-      case meta.Importer(_, List(x))
-          if !(x.is[Importee.Rename] || x.is[Importee.Unimport]) ||
-            style.runner.dialect.allowAsForImportRename &&
-            (ConvertToNewScala3Syntax.enabled ||
-              !x.tokens.exists(_.is[Token.RightArrow])) =>
-        removeToken
-      case _ => null
+    if (ft.left.is[Token.MacroSplice] || ft.left.is[Token.MacroQuote]) {
```

Comment:
quick question: is this legal
```
$/* comment */{...}
```

Reply:
It compiles with scala 3.0.0 so seems it's legal

It fails with scalafmt (both current release and on my branch) with `Symbol literals are no longer allowed`. Looks like the problem lies in `scala.meta.internal.parsers.ScalametaParser`

Reply:
@tgodzik do we want to fix this? if so, `ft.left` should be replaced with `prevNonComment(ft).left`.

Reply:
I can take a look at it after the weekend and but we can for sure update this to work with comments just in case.

Reply:
@note thank you for this submission. rather than torture you again, i made the small change to handle comments when the parser is ready (in #2557).